src/app/visitors/components/visitors-columns.tsx

```diff
@@ -1,10 +1,35 @@
 'use client';
 
 import { ColumnDef } from '@tanstack/react-table';
-import { Visitor } from '@/models/visitor';
+import { Visitor } from '@/models/visitor-schema';
 import { DataTableColumnHeader } from '@/components/data-table-column-header';
+import { Checkbox } from '@/components/ui/checkbox';
 
 export const columns: ColumnDef<Visitor>[] = [
+  {
+    id: 'select',
+    header: ({ table }) => (
+      <Checkbox
+        checked={
+          table.getIsAllPageRowsSelected() ||
+          (table.getIsSomePageRowsSelected() && 'indeterminate')
+        }
+        onCheckedChange={(value) => table.toggleAllPageRowsSelected(!!value)}
+        aria-label='Select all'
+        className='flex translate-y-[2px]'
```

Comment:
I consider adding `flex` here a good option. Why:

When user checks the `Checkbox` element, a span is added to a <button> element with a height of `1 rem` (16px). The height of a button without this span is already 1rem with 1px borders; `display: inline-block` is set. As the `box-sizing: border-box` is set to the button, this 1 rem includes 1px border that make the height of an inner content of `14px` in non-selected state. Adding of `1rem` span inside it causes button to expand inner content area to 16px, adding 2px in height that causes a row also to expand.

Setting `display: flex` together with explicit width and height set allows button not to change its dimensions based on inner content. Content may be not fully visible and can be adjusted via the `overflow` property. But in this case it is not needed.